### PR TITLE
EZP-31118: Introduced strict types for ContentService

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31118](https://jira.ez.no/browse/EZP-31118)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** |  `master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

This PR introduces strict types for the `\eZ\Publish\API\Repository\ContentService`

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [ ] Fix type in commit message before merge
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/API/Repository/ContentService.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ public function loadContentInfoByRemoteId($remoteId);
      *
      * @return \eZ\Publish\API\Repository\Values\Content\VersionInfo
      */
-    public function loadVersionInfo(ContentInfo $contentInfo, $versionNo = null);
+    public function loadVersionInfo(ContentInfo $contentInfo, ?int $versionNo = null): VersionInfo;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/ContentService.php]: docblock for `$versionNo` should be updated to `int|null`

File: eZ/Publish/API/Repository/ContentService.php [hunk 2]
```diff
@@ -94,7 +94,7 @@ public function loadVersionInfo(ContentInfo $contentInfo, $versionNo = null);
      *
      * @return \eZ\Publish\API\Repository\Values\Content\VersionInfo
      */
-    public function loadVersionInfoById($contentId, $versionNo = null);
+    public function loadVersionInfoById(int $contentId, ?int $versionNo = null): VersionInfo;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/ContentService.php]: docblock for `$contentId` and `$versionNo` and should be updated

File: eZ/Publish/API/Repository/ContentService.php [hunk 3]
```diff
@@ -112,7 +112,7 @@ public function loadVersionInfoById($contentId, $versionNo = null);
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Content
      */
-    public function loadContentByContentInfo(ContentInfo $contentInfo, array $languages = null, $versionNo = null, $useAlwaysAvailable = true);
+    public function loadContentByContentInfo(ContentInfo $contentInfo, array $languages = null, ?int $versionNo = null, bool $useAlwaysAvailable = true): Content;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/ContentService.php]: docblock for `$versionNo` should be updated to `int|null`

File: eZ/Publish/API/Repository/ContentService.php [hunk 4]
```diff
@@ -126,7 +126,7 @@ public function loadContentByContentInfo(ContentInfo $contentInfo, array $langua
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Content
      */
-    public function loadContentByVersionInfo(VersionInfo $versionInfo, array $languages = null, $useAlwaysAvailable = true);
+    public function loadContentByVersionInfo(VersionInfo $versionInfo, array $languages = null, bool $useAlwaysAvailable = true): Content;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/ContentService.php]: `@param string[] $language`

File: eZ/Publish/API/Repository/ContentService.php [hunk 5]
```diff
@@ -144,7 +144,7 @@ public function loadContentByVersionInfo(VersionInfo $versionInfo, array $langua
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Content
      */
-    public function loadContent($contentId, array $languages = null, $versionNo = null, $useAlwaysAvailable = true);
+    public function loadContent(int $contentId, array $languages = null, ?int $versionNo = null, bool $useAlwaysAvailable = true): Content;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/ContentService.php]: docblock for `$contentId` and `$versionNo` and should be updated

File: eZ/Publish/API/Repository/ContentService.php [hunk 6]
```diff
@@ -229,7 +229,7 @@ public function updateContentMetadata(ContentInfo $contentInfo, ContentMetadataU
      *
      * @return mixed[] Affected Location Id's (List of Locations of the Content that was deleted)
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/ContentService.php]:
```suggestion
     * @return int[] Affected Location Id's (List of Locations of the Content that was deleted)
``` 
?

File: eZ/Publish/API/Repository/ContentService.php [hunk 7]
```diff
@@ -246,7 +246,7 @@ public function deleteContent(ContentInfo $contentInfo);
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Content - the newly created content draft
      */
-    public function createContentDraft(ContentInfo $contentInfo, VersionInfo $versionInfo = null, User $creator = null);
+    public function createContentDraft(ContentInfo $contentInfo, ?VersionInfo $versionInfo = null, ?User $creator = null): Content;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/ContentService.php]: docblock for `$versionInfo` and `$creator`and should be updated

File: eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php [hunk 8]
```diff
@@ -186,7 +186,7 @@ public function insertContentObject(CreateStruct $struct, $currentVersionNo = 1)
 
         $q->prepare()->execute();
 
-        return $this->dbHandler->lastInsertId(
+        return (int)$this->dbHandler->lastInsertId(
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php]: those are type-casting bugs to be fixed for 7.5, unfortunately. Could be added to #2843.

File: eZ/Publish/Core/Repository/ContentService.php [hunk 9]
```diff
@@ -153,12 +151,12 @@ public function loadContentInfoList(array $contentIds): iterable
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException - if the content with the given id does not exist
      *
-     * @param mixed $id
+     * @param string|int $id
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/ContentService.php]: hmm, this needs to be refactored to get rid of this nonsense. Two separate methods, with reusable parts, if possible.
IMHO it falls into the scope of introducing strict types in this case.

File: eZ/Publish/Core/Repository/ContentService.php [hunk 10]
```diff
@@ -326,7 +324,7 @@ public function loadContent($contentId, array $languages = null, $versionNo = nu
      * @internal
      * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if the content or version with the given id and languages does not exist
      *
-     * @param mixed $id
+     * @param string|int $id
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/ContentService.php]: same comment as above

**Comment by adamwojs** [comment]
PR updated according to @mikadamczyk code review suggestions.

**Comment by adamwojs** [comment]
PR updated according to @alongosz code review suggestions.

**Pull request merged**